core/src/main/java/bisq/core/dao/state/storage/BlocksPersistence.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.dao.state.storage;
+
+import bisq.common.file.FileUtil;
+import bisq.common.proto.persistable.PersistenceProtoResolver;
+
+import protobuf.BaseBlock;
+
+import java.nio.file.Path;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nullable;
+
+@Slf4j
+public class BlocksPersistence {
+    // 10000->Writing 130014 blocks took 1658 msec
+    // 1000-> Writing 130014 blocks took 1685 msec  Mapping blocks from DaoStateStore took 2250 ms
+    public static final int BUCKET_SIZE = 1000; // results in about 1 MB files and about 1 new file per week
+
+    private final File storageDir;
+    private final String fileName;
+    private final PersistenceProtoResolver persistenceProtoResolver;
+    private Path usedTempFilePath;
+
+    public BlocksPersistence(File storageDir, String fileName, PersistenceProtoResolver persistenceProtoResolver) {
+        this.storageDir = storageDir;
+        this.fileName = fileName;
+        this.persistenceProtoResolver = persistenceProtoResolver;
+
+       /* if (!storageDir.exists()) {
+            storageDir.mkdir();
+        }*/
```

Comment:
Why is this commented out?

Reply:
I guess it can be safely removed as it is checked during the write procedure anyways.

Reply:
Got enabeld in a later commit again as it was required at some cases (resync from resources i guess, or regtest fresh start)